src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAddressService.java

```diff
@@ -55,7 +65,57 @@ public AdminCourtAddressService(final CourtRepository courtRepository, final Cou
             .collect(toList());
     }
 
-    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    public List<String> validateCourtAddressPostcodes(final String slug, final List<CourtAddress> courtAddresses) {
+        final List<String> invalidPostcodes = new ArrayList<>();
+
+        if (!CollectionUtils.isEmpty(courtAddresses)) {
+            final List<String> allPostcodes = getAllPostcodesSortedByAddressType(courtAddresses);
+
+            if (!CollectionUtils.isEmpty(allPostcodes)) {
+                invalidPostcodes.addAll(validationService.validateFullPostcodes(allPostcodes));
+            }
+
+            // Only update the in-person court coordinates is all input postcodes are valid
+            if (invalidPostcodes.isEmpty()
+                && !CollectionUtils.isEmpty(allPostcodes)
+                && isInPersonCourt(slug)) {
+                updateCourtLatLonUsingPrimaryPostcode(slug, allPostcodes.get(0));
+            }
+        }
+        return invalidPostcodes;
+    }
+
+    private List<String> getAllPostcodesSortedByAddressType(final List<CourtAddress> courtAddresses) {
+        final Map<Integer, uk.gov.hmcts.dts.fact.entity.AddressType> addressTypeMap = addressTypeService.getAddressTypeMap();
+
+        // Sort the postcodes so that the 'visit us' or 'visit or contact us' postcodes appear first
+        return courtAddresses.stream()
+            .sorted(Comparator.comparingInt(a -> AddressType.isCourtAddress(getAddressTypeFromId(addressTypeMap, a.getAddressTypeId())) ? 0 : 1))
+            .map(CourtAddress::getPostcode)
+            .filter(StringUtils::isNotBlank)
+            .collect(toList());
+    }
+
+    private String getAddressTypeFromId(final Map<Integer, uk.gov.hmcts.dts.fact.entity.AddressType> map, final Integer addressTypeId) {
+        if (!map.containsKey(addressTypeId)) {
+            throw new IllegalArgumentException("Unknown address type ID: " + addressTypeId);
+        }
+        return map.get(addressTypeId).getName();
+    }
+
+    private boolean isInPersonCourt(final String slug) {
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+        return courtEntity.getInPerson() == null || courtEntity.getInPerson().getIsInPerson();
```

Comment:
Is it possible for the getInPerson to be empty too? i.e 

courtEntity.getInPerson() == null || courtEntity.getInPerson().isEmpty()

Probably not I assume, thought I'd ask though

Reply:
1. it is not possible to have courtEntity.getInPerson().isEmpty(). It is not s string but an object
2. There are gaps in the database table. Certain courts are missing on the in-person table and if that happens, they will be considered in-person courts for the front end and have a direction. So this is to provide compatibility with the user front end. Can provide a comment here but this logic needs to remain